### ISSUE TITLE
SC2037: False positive

### ISSUE THREAD

**Issue by Marek77** [koalaman/shellcheck#876]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2037
- My shellcheck version (`shellcheck --version` or "online"): 0.4.5
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```
#!/bin/bash

PDF_SHA1_TAG='abcdef01'

COMMIT_LOG=$(PAGER=cat git log --abbrev=8 --oneline "$PDF_SHA1_TAG"~..HEAD)
echo "$COMMIT_LOG"
```

#### Here's what shellcheck currently says:
/tmp/s:5:14: warning: To assign the output of a command, use var=$(cmd) . [SC2037]

#### Here's what I wanted or expected to see:
No such message; this is a false positive.

**Comment by Nightfirecat** [comment]
How much of the offending line can you remove and still cause it to trigger that warning? I'm assuming it's the `"$PDF_SHA1_TAG"~..HEAD` part of it that's tripping it up. Just curious to see the most minimal case here.

**Comment by koalaman** [comment]
You're right, this is a definitely a fixable false positive.

The minimal case would be e.g. `PAGER=cat mail`.

SC2037 checks for prefix assignments that only contain a common command, thinking that the user wanted to run `cat mail` and assign it, instead of running `mail` with `PAGER=cat` in the environment.

The real annoyance is not the fact that `PAGER` is commonly used in this way, but that `PAGER="cat" mail` still triggers, even though the user's intention is clear. That should be fixed.

**Comment by koalaman** [comment]
This has been fixed in faafc99. Thanks!

It will suggest that you quote `PAGER="cat" git log ..` though.
